server/src/main/java/io/crate/session/Session.java

```diff
@@ -713,11 +713,13 @@ private CompletableFuture<?> triggerDeferredExecutions() {
 
     private CompletableFuture<?> exec(List<DeferredExecution> executions) {
         if (executions.size() == 1) {
-            var toExec = executions.get(0);
-            return singleExec(toExec.portal(), toExec.resultReceiver(), toExec.maxRows());
-        } else {
-            return bulkExec(executions);
+            var toExec = executions.getFirst();
+            ResultReceiver<?> resultReceiver = toExec.resultReceiver();
+            if (!resultReceiver.isBulk()) {
+                return singleExec(toExec.portal(), toExec.resultReceiver(), toExec.maxRows());
+            }
```

Comment:
Couldn't we instead change `sync` to have a `bool forceBulk` parameter? Seems a bit clunky to tie the execution for all row receivers based on a property on in the first one.